From a7c53e0c2cd8fc45eac90c0468e44697019ceda6 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Sat, 24 Mar 2018 12:15:43 -0400
Subject: fix out-of-tree build of crt files with stack protector enabled

the makefile logic for these files was wrong in the out-of-tree case,
but it likely only affected the "all" level of stack protector.
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 308ddaa..46fc91e 100644
--- a/Makefile
+++ b/Makefile
@@ -119,7 +119,7 @@ $(OPTIMIZE_SRCS:$(srcdir)/%.c=obj/%.o) $(OPTIMIZE_SRCS:$(srcdir)/%.c=obj/%.lo):
 MEMOPS_SRCS = src/string/memcpy.c src/string/memmove.c src/string/memcmp.c src/string/memset.c
 $(MEMOPS_SRCS:%.c=obj/%.o) $(MEMOPS_SRCS:%.c=obj/%.lo): CFLAGS_ALL += $(CFLAGS_MEMOPS)
 
-NOSSP_SRCS = $(wildcard crt/*.c) \
+NOSSP_SRCS = $(patsubst $(srcdir)/%,%,$(wildcard $(srcdir)/crt/*.c)) \
 	src/env/__libc_start_main.c src/env/__init_tls.c \
 	src/env/__stack_chk_fail.c \
 	src/thread/__set_thread_area.c src/thread/$(ARCH)/__set_thread_area.c \
-- 
cgit v0.11.2

